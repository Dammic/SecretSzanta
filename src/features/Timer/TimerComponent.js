import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { Button } from '../Shared/Buttons'

import styles from './Timer.module.css'

const TimerComponent = ({ secondsRemaining, isVetoUnlocked, onVetoClick }) => (
    <div className={classNames(styles.timer, { [styles.visible]: !!secondsRemaining })}>
        <strong>{secondsRemaining}</strong>s remaining
        {isVetoUnlocked && <Button className={styles.veto} onClick={onVetoClick}>veto</Button>}
    </div>
)

TimerComponent.displayName = 'TimerComponent'
TimerComponent.propTypes = {
    secondsRemaining: PropTypes.number,
    onVetoClick: PropTypes.func,
    isVetoUnlocked: PropTypes.bool,
}

export default TimerComponent
