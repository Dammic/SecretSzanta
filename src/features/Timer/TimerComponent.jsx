import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const TimerComponent = ({ secondsRemaining, isVetoUnlocked, onVetoClick }) => {
    return (
        <div className={classNames('timer', { visible: secondsRemaining })}>
            <strong>{secondsRemaining}</strong>s remaining
            {isVetoUnlocked && <a role="button" tabIndex="0" className="btn veto" onClick={onVetoClick}>veto</a>}
        </div>
    )
}

TimerComponent.propTypes = {
    secondsRemaining: PropTypes.number,
    onVetoClick: PropTypes.func,
    isVetoUnlocked: PropTypes.bool,
}

export default TimerComponent
