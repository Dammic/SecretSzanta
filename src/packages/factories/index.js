export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
