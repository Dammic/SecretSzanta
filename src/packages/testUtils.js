import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
