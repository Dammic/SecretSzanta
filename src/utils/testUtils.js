import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
