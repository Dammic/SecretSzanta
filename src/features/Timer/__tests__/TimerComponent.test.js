import React from 'react'
import { shallow } from 'enzyme'
import { expectMatchingSnapshot } from 'packages/testUtils'
import TimerComponent from '../TimerComponent'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        secondsRemaining: 0,
        isVetoUnlocked: false,
        onVetoClick: jest.fn(),
        ...propsOverrides,
    }
    const component = renderMethod(<TimerComponent {...props} />)
    return { props, component }
}

describe('<TimerComponent />', () => {
    it('timer is hidden when there are no seconds remaining and veto button is not rendered', () => {
        const { props } = setupProps({ secondsRemaining: 0, isVetoUnlocked: false })
        expectMatchingSnapshot(<TimerComponent {...props} />)
    })

    it('shows timer when there are remaining seconds and veto button is not rendered', () => {
        const { props } = setupProps({ secondsRemaining: 10, isVetoUnlocked: false })
        expectMatchingSnapshot(<TimerComponent {...props} />)
    })

    it('shows timer when there are remaining seconds and veto button is rendered if there is veto possibility', () => {
        const { props } = setupProps({ secondsRemaining: 10, isVetoUnlocked: true })
        expectMatchingSnapshot(<TimerComponent {...props} />)
    })
})
