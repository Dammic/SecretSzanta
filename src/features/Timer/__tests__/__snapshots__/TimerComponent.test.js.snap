// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerComponent /> shows timer when there are remaining seconds and veto button is not rendered 1`] = `
<div
  className="timer visible"
>
  <strong>
    10
  </strong>
  s remaining
</div>
`;

exports[`<TimerComponent /> shows timer when there are remaining seconds and veto button is rendered if there is veto possibility 1`] = `
<div
  className="timer visible"
>
  <strong>
    10
  </strong>
  s remaining
  <div
    className="btn veto"
    onClick={[MockFunction]}
    role="button"
    tabIndex="0"
  >
    veto
  </div>
</div>
`;

exports[`<TimerComponent /> timer is hidden when there are no seconds remaining and veto button is not rendered 1`] = `
<div
  className="timer"
>
  <strong>
    0
  </strong>
  s remaining
</div>
`;
